addons/point_of_sale/models/pos_session.py

```diff
@@ -955,6 +955,7 @@ def _create_combine_account_payment(self, payment_method, amounts, diff_amount):
             'ref': _('Combine %s POS payments from %s', payment_method.name, self.name),
             'pos_payment_method_id': payment_method.id,
             'pos_session_id': self.id,
+            'company_id': self.company_id.id,
```

Comment:
This solved another related issues that occurred with multi-company.
If we were closing a branch session while the parent was also selected, the payment company_id was set to the parent id.